neonvm/controllers/vm_controller.go

```diff
@@ -374,12 +374,15 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 		vm.Status.Phase = vmv1.VmPending
 	case vmv1.VmPending:
 		// Generate runner pod name and set desired memory provider.
-		// Together with Status.MemoryProvider set for PodName != "" above,
-		// It is now guaranteed to have Status.MemoryProvider != nil
 		if len(vm.Status.PodName) == 0 {
 			vm.Status.PodName = names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", vm.Name))
+			//nolint:staticcheck // We know it's deprecated. That's why we're checking it.
```

Comment:
Do we really need this check, or we can just leave MemoryProvider in whatever state, as long as it is not nil?

Reply:
Good point, yeah. Fixed in a306ee16a7642c779367f90551d89cc5afc77de2